eng/common/pipelines/templates/steps/credscan.yml

```diff
@@ -9,7 +9,12 @@ steps:
     if ("$(Build.Reason)" -eq 'PullRequest') {
       $targetBranch = "origin/$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
       $changedFiles = git diff $targetBranch HEAD --name-only --diff-filter=d
-      $changedFiles | ForEach-Object { Add-Content -Path "${{ parameters.SourceDirectory }}/credscan.tsv" -Value "${{ parameters.SourceDirectory }}/$_"}
+      foreach ($filePath in $changedFiles){
+        if ($filePath-match [System.IO.Path]::GetInvalidPathChars()) {
+          $filePath = Split-Path -Path $filePath
```

Comment:
I would have expected an enumeration error to occur here since we are modifying the thing we are using to enumerate the collection.

Reply:
Powershell did not have issues on modification of enumeration elements. It is another copy of the variable.

Reply:
Add-Content here is actually appending to the file so we aren't modifying an in-memory file list.

Reply:
However based on the other comment we probably should start creating a list.

Reply:
> Add-Content here is actually appending to the file so we aren't modifying an in-memory file list.

I was referring to L14 where we set the $filePath variable, but Sima mentioned that it is a different copy of the variable.